### PR TITLE
88 get network

### PR DESCRIPTION
**Pull request by edavalosanaya**
Did the following:

1. Made the test workflow only run for pushes or PR to the main branch.
2. Added the route to the Manager
3. Added testing of the new route, refactored `test_p2p_networking.py` to re-use configured manager fixtures, and avoid repetition of these pytest fixtures.

### CODE REVIEW

File: test/networking/test_p2p_networking.py [hunk 1]
```diff
@@ -11,158 +11,7 @@
...
-    assert manager.commit_graph(graph, {"local": ["Gen1"], "local2": ["Con1"]})
-
-    yield manager
-
-    worker1.shutdown()
-    worker2.shutdown()
-
-
-#@pytest.mark.repeat(10)
+# @pytest.mark.repeat(10)
```
Comment (umesh-timalsina) [inline test/networking/test_p2p_networking.py:14]: Not sure why this was added.

**Review by umesh-timalsina — APPROVED** [review]
Overall looks good. A single comment added. Also, not sure if we want to include the GHA(a separate concern) in this PR.

**Comment by umesh-timalsina** [comment]
On the examples/remote_camera.py, I am getting the following error:

```python
Traceback (most recent call last):
  File "/home/umesh/micromamba/envs/chimerapy-dev/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
  File "/home/umesh/micromamba/envs/chimerapy-dev/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/home/umesh/isis/oele-isis/ChimeraPy/chimerapy/manager.py", line 107, in get_network
    return web.json_response(self.dashboard_dict())
  File "/home/umesh/isis/oele-isis/ChimeraPy/chimerapy/manager.py", line 179, in dashboard_dict
    "ip": self.nodes_server_table[node_name]["host"],
KeyError: 'web'

```

**Comment by umesh-timalsina** [comment]
> On the [`examples/remote_camera.py`](https://github.com/oele-isis-vanderbilt/ChimeraPy/blob/81b508b1c2f27cc4a74ca050e3c0819fd8bd699d/examples/remote_camera.py), I am getting the following error:
> 
> ```python
> Traceback (most recent call last):
>   File "/home/umesh/micromamba/envs/chimerapy-dev/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
>     resp = await request_handler(request)
>   File "/home/umesh/micromamba/envs/chimerapy-dev/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
>     resp = await handler(request)
>   File "/home/umesh/isis/oele-isis/ChimeraPy/chimerapy/manager.py", line 107, in get_network
>     return web.json_response(self.dashboard_dict())
>   File "/home/umesh/isis/oele-isis/ChimeraPy/chimerapy/manager.py", line 179, in dashboard_dict
>     "ip": self.nodes_server_table[node_name]["host"],
> KeyError: 'web'
> ```
> 
> Last two lines from the console (stdout/err):
> 
> ```
> 2023-02-08 12:52:47 [DEBUG] chimerapy: <Manager @129.59.104.153:9001>: receiving Worker's node server request: SUCCESS
> Ready to start? (Y/n)
> ```

This might be happening because of the table not being updated at the time of the request. As this will eventually be event-driven, it might make sense to make it such that, we can send incomplete info but periodically update the client on network changes.

**Comment by edavalosanaya** [comment]
Good check, this error is happening because the GET request occurred after the Node is created but before the Node finished connecting and reported its IP and port data to the Manager.

**Pull request merged**